src/cards/polymorph.ts

```diff
@@ -0,0 +1,194 @@
+import { CardCategory, UnitType } from '../types/commonTypes';
+import { playDefaultSpellSFX } from './cardUtils';
+import { Spell, addTarget, allModifiers, getCurrentTargets, refundLastSpell } from './index';
+import { CardRarity, probabilityMap } from '../types/commonTypes';
+import * as Unit from '../entity/Unit';
+import * as Pickup from '../entity/Pickup';
+import * as Image from '../graphics/Image';
+import { allUnits } from '../entity/units';
+import { isModActive } from '../registerMod';
+import Underworld from '../Underworld';
+import { IPickup, isPickup, pickups } from '../entity/Pickup';
+import { bossmasonUnitId } from '../entity/units/deathmason';
+import floatingText from '../graphics/FloatingText';
+import seedrandom from 'seedrandom';
+import { chooseObjectWithProbability, chooseOneOf, chooseOneOfSeeded, getUniqueSeedString } from '../jmath/rand';
+
+export const polymorphId = 'Polymorph';
+const spell: Spell = {
+  card: {
+    id: polymorphId,
+    category: CardCategory.Curses,
+    supportQuantity: true,
+    manaCost: 40,
+    healthCost: 0,
+    expenseScaling: 2,
+    probability: probabilityMap[CardRarity.RARE],
+    thumbnail: 'spellIconPolymorph.png',
+    sfx: 'purify',
+    description: ['spell_polymorph'],
+    effect: async (state, card, quantity, underworld, prediction) => {
+      const targets = getCurrentTargets(state);
+      if (targets.length == 0) {
+        refundLastSpell(state, prediction, 'No valid targets. Cost refunded.');
+        return state;
+      }
+
+      playDefaultSpellSFX(card, prediction);
+
+      for (let i = 0; i < quantity; i++) {
+        for (const target of targets) {
+          if (Unit.isUnit(target)) {
+            const unit = target as Unit.IUnit;
```

Comment:
This typecast `as Unit.IUnit` _should_ be unnecessary due to the [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) in the line just above it